packages/state/foundry/src/eventMappers.ts

```diff
@@ -94,38 +96,38 @@ export function getPresenceEvent(
   documentSchema: DocumentSchema,
   foundryUpdate: PresenceCollaborativeUpdate,
 ): PresenceEvent {
-  switch (foundryUpdate.type) {
-    case "presenceChangeEvent": {
-      const { userId, status } = foundryUpdate.presenceChangeEvent;
-      const eventData = status === "PRESENT" ? ARRIVED_DATA : DEPARTED_DATA;
-      return {
-        eventData,
-        userId: userId as UserId,
-      };
-    }
-
-    case "customPresenceEvent": {
-      const { userId, eventData } = foundryUpdate.customPresenceEvent;
-      const presenceEventData = getPresenceEventData(documentSchema, eventData);
-      return {
-        eventData: presenceEventData,
-        userId: userId as UserId,
-      };
-    }
-    default: {
-      foundryUpdate satisfies never;
-      const unknownUpdate = foundryUpdate as Record<string, unknown>;
-      return {
-        eventData: {
-          rawData: foundryUpdate,
-          rawType: typeof unknownUpdate.type === "string" ? unknownUpdate.type : "unknown",
-          type: PresenceEventDataType.UNKNOWN,
-        },
-        // FIXME: try and pull userId from message? Or just fix the platform types to have useful wrappers.
-        userId: invalidUserRef().userId,
-      };
-    }
+  if (foundryUpdate.type === "presenceChangeEvent") {
+    const event = foundryUpdate as { type: "presenceChangeEvent" } & DocumentPresenceChangeEvent;
```

Comment:
It appears that the native types are harder to discriminate automatically in typescript, so we may prefer to keep the FE types as is.